### PR TITLE
Add smt's keymap for Preonic

### PR DESCRIPTION
**Pull request by smt**
Adding my Preonic keymap.

### CODE REVIEW

File: keyboards/preonic/keymaps/smt/keymap.c [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+#define _DVORAK 0
+#define _QWERTY 1
+#define _LOWER 2
+#define _RAISE 3
+#define _ADJUST 16
+
+enum preonic_keycodes {
+  DVORAK = SAFE_RANGE,
+  QWERTY,
+  COLEMAK,
```
Comment (smt) [inline keyboards/preonic/keymaps/smt/keymap.c]: @jackhumbert I ditched the Colemak layer, but the firmware wouldn't run properly without keeping this in the enum. What is the best way to safely remove it?

Comment (jackhumbert) [inline keyboards/preonic/keymaps/smt/keymap.c]: What kind of error/problem are you seeing? I don't know of anything that would require that specifically. The persistent default layer is stored as a number, though, if that's what is acting up.

Comment (smt) [inline keyboards/preonic/keymaps/smt/keymap.c]: If I remove COLEMAK, the build and flash seem to go ok, but the keyboard is unresponsive after starting up. Once that happened, I built the default image, flashed, and incrementally added my changes back in until I narrowed it down to this line.

It is in a working state like this, for now, and when I have a moment later today, I'll continue to play with it. It's an opportunity to introduce myself to the debugging procedure, but I have never done anything with C before now 😄 

For context, I have been using your docker image to build, and my workflow is pretty much this:

```
$ rm ./preonic_smt.hex
$ docker run \
  -e keymap=smt -e keyboard=preonic -e subproject='' \
  --rm -v $(pwd):/qmk:rw jackhumbert/qmk_firmware make clean
$ docker run \
  -e keymap=smt -e keyboard=preonic -e subproject='' \
  --rm -v $(pwd):/qmk:rw jackhumbert/qmk_firmware make
<put keyboard into flash mode>
$ dfu-programmer atmega32u4 erase
$ dfu-programmer atmega32u4 flash ./preonic_smt.hex
$ dfu-programmer atmega32u4 start
```

Comment (jackhumbert) [inline keyboards/preonic/keymaps/smt/keymap.c]: Can you try changing your keyboard's default layer to 0 (or qwerty) before flashing?

Comment (smt) [inline keyboards/preonic/keymaps/smt/keymap.c]: @jackhumbert Heh, tried again and it worked! I'll commit my change. There must have been an error in my process.

Comment (jackhumbert) [inline keyboards/preonic/keymaps/smt/keymap.c]: Awesome :)

**Comment by jackhumbert** [comment]
Awesome! Thanks :)

**Pull request merged**